internal/pkg_npm/pkg_npm.bzl

```diff
@@ -295,11 +295,15 @@ def _pkg_npm(ctx):
         # Only collect DefaultInfo files (not transitive)
         deps_files_depsets.append(dep.files)
 
-        # All direct & transitive JavaScript-producing deps
-        if JSModuleInfo in dep:
-            deps_files_depsets.append(dep[JSModuleInfo].sources)
+        # All direct and transitive deps that produce CommonJS modules
+        if JSNamedModuleInfo in dep:
```

Comment:
I'm concerned about dropping `JsModuleInfo` - I'm pretty sure it's possible that we get a dep that has JS files only in that provider but not in either of these two you've added. I think we should have all three. Would be nice if there was a red test result on this PR making me more confident...

Reply:
I've added the `JSModuleInfo` back, but I'm not sure where this provider is created. If you could give me an example of a rule where it's used, I could add a test case with it as well.